packages/dashboard/src/components/internalDashboard/collapsiblePanel.tsx

```diff
@@ -1,19 +1,95 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
 
 import './index.css';
-import { colorBackgroundButtonPrimaryDefault, spaceContainerHorizontal } from '@cloudscape-design/design-tokens';
+import {
+  colorBackgroundButtonPrimaryDefault,
+  colorBorderDividerDefault,
+  spaceContainerHorizontal,
+  spaceStaticXxxs,
+} from '@cloudscape-design/design-tokens';
+import Box, { BoxProps } from '@cloudscape-design/components/box';
+import Button from '@cloudscape-design/components/button';
+import { IconProps } from '@cloudscape-design/components/icon';
+import { DEFAULT_COLLAPSED_SIDE_PANE_WIDTH } from '../resizablePanes/constants';
 
-export function CollapsiblePanel(props: { icon: string; dataCy: string }) {
-  return (
+type CollapsiblePanelProps = {
+  isPanelCollapsed: boolean;
+  panelWidth: number;
+  onCollapsedPanelClick: () => void;
+  panelContent: ReactNode;
+  icon: string;
+  side: 'right' | 'left';
+  headerText: string;
+};
+
+export function CollapsiblePanel(props: CollapsiblePanelProps) {
+  // Used to set the borderRight or borderLeft prop of the pane
+  const borderSide = props.side === 'right' ? 'Left' : 'Right';
```

Comment:
if we made the borderSide lowercase would we need to cast the iconside?

Reply:
`borderSide` has to better capitalized because the value it's used for must be `borderRight` or `borderLeft`. All lower case doesn't work. I'm just reusing the fact that this is the flipped side for the `iconSide` variable.